pandas/io/pytables.py

```diff
@@ -3185,6 +3186,8 @@ def write_array(
         #  both self._filters and EA
 
         value = extract_array(obj, extract_numpy=True)
+        if isinstance(value, BaseStringArray):
```

Comment:
Should this logic live in `extract_array`?

Reply:
The line from the docstring of `extract_array` states:

    Extract the ndarray or ExtensionArray from a Series or Index.

and the `extract_numpy` argument:

    Whether to extract the ndarray from a NumpyExtensionArray.

So I think no - I would expect that function to still return the ExtensionArray.